framework/test/jest/integration/specs/modules/chain/components/storage/entities/Account/decreaseFieldBy.spec.js

```diff
@@ -0,0 +1,170 @@
+const {
+	Account,
+} = require('../../../../../../../../../../src/modules/chain/components/storage/entities');
+const { PgHelper } = require('../../../../../../../utils/pg-helper');
+
+describe('storage.entities.Account.decreaseFieldBy', () => {
+	let pgHelper;
+	let storage;
+	let db;
+
+	beforeAll(async () => {
+		// Arrange
+		pgHelper = new PgHelper({ dbName: 'AccountDecreaseFieldBy' });
+
+		// Create second postgres connection
+		db = await pgHelper.bootstrap();
+
+		// Setup storage for Accounts
+		storage = await pgHelper.createStorage();
+		storage.registerEntity('Account', Account);
+
+		storage.entities.Account.extendDefaultOptions({
+			limit: 101, // @todo get it from constants
+		});
+	});
+
+	afterAll(async () => {
+		await db.done();
+		await pgHelper.cleanup();
+	});
+
+	beforeEach(async () => {
+		await db.query('DELETE FROM mem_accounts');
+	});
+
+	describe('Given arguments ({publicKey}, "producedBlocks", "1", tx)', () => {
+		it('should decrease "producedBlocks" property of given account by "1"', async () => {
+			// Arrange
+			const account = {
+				address: 'delegateAddress',
+				isDelegate: 0,
+				publicKey:
+					'399a7d14610c4da8800ed929fc6a05133deb8fbac8403dec93226e96fa7590ee',
+				balance: 100,
+				producedBlocks: 5,
+			};
+
+			const expectedValue = 4;
+
+			await PgHelper.createAccount(db, account);
+
+			// Act
+			await db
+				.tx(async tx => {
+					await storage.entities.Account.decreaseFieldBy(
+						{
+							publicKey: account.publicKey,
+						},
+						'producedBlocks',
+						'1',
+						tx,
+					);
+				})
+				.catch(console.error);
+
+			const updatedAccount = await PgHelper.getAccountByPublicKey(
+				db,
+				account.publicKey,
+			);
+
+			// Assert
+			expect(updatedAccount.producedBlocks).toEqual(expectedValue);
```

Comment:
Hmm. This is an interesting question. I'd say no because we provide a field name and updating that field is what decreaseFieldBy is promising to do. As long as it's doing that rest shouldn't matter. But let's discuss this tomorrow with a quick call to see if there is a case that another field could be updated with this?